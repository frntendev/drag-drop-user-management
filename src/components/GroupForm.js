import React from "react";
import { Field } from "redux-form";
import { css } from "react-emotion";
import TextField from "material-ui/TextField";
const renderTextField = ({
  input,
  label,
  meta: { touched, error },
  ...custom
}) => (
  <TextField
    id={label}
    label={label}
    error={touched && error ? true : false}
    {...input}
    {...custom}
  />
);
const required = value => (value ? undefined : "Required");

const GroupForm = props => {
  return (
    <div
      id="group-form"
      className={css`
        display: flex;
        flex-wrap: wrap;
      `}
    >
      <Field
        name="name"
        component={renderTextField}
        label="Name"
        validate={[required]}
        className={css`
          flex: 1;
          margin-right: 5%;
        `}
      />
      <Field
        name="location"
        component={renderTextField}
        label="Location"
        validate={[required]}
        className={css`
          flex: 1;
        `}
      />
    </div>
  );
};

export default GroupForm;
