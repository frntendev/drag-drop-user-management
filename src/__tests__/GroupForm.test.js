import React from "react";
import { shallow, mount, configure } from "enzyme";
import { Field } from "redux-form";
import GroupForm from "../components/GroupForm";
import Adapter from "enzyme-adapter-react-15";
import TextField from "material-ui/TextField";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });
const renderTextField = ({
  input,
  label,
  meta: { touched, error },
  ...custom
}) => (
  <TextField
    label={label}
    error={touched && error ? true : false}
    {...input}
    {...custom}
  />
);
describe("GroupForm Component", () => {
  const shallowComponent = shallow(<GroupForm />);
  it("should render without throwing an error", () => {
    expect(shallowComponent).toMatchSnapshot();
  });
  it("should find textfields in the component", () => {
    expect(shallowComponent.find("#group-form").length).toEqual(1);
    expect(shallowComponent.find("[name='name']").length).toEqual(1);
    expect(shallowComponent.find("[name='location']").length).toEqual(1);
  });
});
