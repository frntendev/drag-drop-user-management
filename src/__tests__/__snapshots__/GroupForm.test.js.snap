// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupForm Component should render without throwing an error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroupForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Field
          className="css-1ose7zb"
          component={[Function]}
          label="Name"
          name="name"
          validate={
            Array [
              [Function],
            ]
          }
        />,
        <Field
          className="css-r6z50o"
          component={[Function]}
          label="Location"
          name="location"
          validate={
            Array [
              [Function],
            ]
          }
        />,
      ],
      "className": "css-ak0z00",
      "id": "group-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "css-1ose7zb",
          "component": [Function],
          "label": "Name",
          "name": "name",
          "validate": Array [
            [Function],
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "css-r6z50o",
          "component": [Function],
          "label": "Location",
          "name": "location",
          "validate": Array [
            [Function],
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Field
            className="css-1ose7zb"
            component={[Function]}
            label="Name"
            name="name"
            validate={
              Array [
                [Function],
              ]
            }
          />,
          <Field
            className="css-r6z50o"
            component={[Function]}
            label="Location"
            name="location"
            validate={
              Array [
                [Function],
              ]
            }
          />,
        ],
        "className": "css-ak0z00",
        "id": "group-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "css-1ose7zb",
            "component": [Function],
            "label": "Name",
            "name": "name",
            "validate": Array [
              [Function],
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "css-r6z50o",
            "component": [Function],
            "label": "Location",
            "name": "location",
            "validate": Array [
              [Function],
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
